backend/src/game/game.gateway.ts

```diff
@@ -0,0 +1,159 @@
+import {
+	SubscribeMessage,
+	MessageBody,
+	WebSocketGateway,
+	WebSocketServer,
+	OnGatewayInit,
+	OnGatewayConnection,
+	OnGatewayDisconnect,
+  } from '@nestjs/websockets';
+import { Logger } from '@nestjs/common';
+import { Socket, Namespace } from 'socket.io';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { UserService } from 'src/user/user.service';
+import { GameService } from './game.service';
+import { User, Match } from '@prisma/client'
+
+@WebSocketGateway({
+	namespace: 'game',
+	cors: {
+	  origin: 'http://localhost:3000', // Update with your client's origin
+	  methods: ['GET', 'POST'],
+	  transports: ['websocket'],
+	  credentials: true,
+	},
+})
+export class GameGateway
+implements OnGatewayInit, OnGatewayConnection, OnGatewayDisconnect
+{
+	// private paddleRightY: number = 250;
+	// private paddleLeftY: number = 250;
+	constructor(
+        private prisma: PrismaService,
+        private readonly userService: UserService,
+		private readonly gameService: GameService
+    ) {}
+	@WebSocketServer() server: Namespace;
+	private logger: Logger = new Logger('GameGateway');
+
+	// @SubscribeMessage('message')
+	// handleMessage(@MessageBody() data: { sender: string; message: string }) {
+	//   this.logger.log(`Message received from ${data.sender}: ${data.message}`);
+	//   // Broadcast the message to all connected clients
+	//   this.server.emit('message', data);
+	// }
+
+	@SubscribeMessage('createNewGame')
+	async handleNewGame(client: any, socketID: string) {
+		const newGame: Match = await this.gameService.newGame(socketID)
+		this.server.emit('newGame', {
+			game: newGame,
+        });
+	}
+
+	@SubscribeMessage('acceptGame')
+	async handleNewGameAccept(client: any, gameID: string, socketID: string) {
```

Comment:
Deze logica kan ook gedaan worden met een POST request (bijvoorbeeld POST http://localhost/game/acceptGame/:gameID/:token). 
Je gebruikt hier niet de websocket om in real time iets bij een andere user te updaten. Dus dan zou ik het altijd via een POST request doen. 
Is discutabel, want zo werkt het ook. Het is denk ik gewoon mooiere code als je HTTP request gebruikt waar kan.